### ISSUE TITLE
Date formats not correct

### ISSUE THREAD

**Issue by jdanthinne** [backdrop-contrib/date#2]
I think all return values from _system_get_date_formats()_ class (there are a lot of them in this module) should be handled differently, because they don't return the same values as before, and this module is unusable right now (see Default values in the field admin ie.).

**Comment by Graham-72** [comment]
I see what you mean. I will work on it.

**Comment by Graham-72** [comment]
I have now made an alteration to function date_format_type_options in date/date_api/date_api.module but there seem to be other issues and maybe this is a whole can of worms!

I note that system_get_date_formats() produces quite different results in Backdrop compared with Drupal7 - see images.
In Backdrop:
![backdrop_systemdateformats](https://cloud.githubusercontent.com/assets/10559274/6935787/939c14ba-d842-11e4-8dcc-71e43878c4b1.jpg)
And in Drupal 7:
![drupal7_systemdates](https://cloud.githubusercontent.com/assets/10559274/6935797/a182786c-d842-11e4-9a35-27853f887bff.jpg)

Other related issues: 
backdrop/backdrop-issues/issues/838
backdrop/backdrop-issues/issues/804
backdrop/backdrop-issues/issues/709
backdrop-contrib/date#1
backdrop-ops/contrib#76
https://api.backdropcms.org/node/41663

**Comment by jdanthinne** [comment]
Yes, I also thought this was a simple change to do, but since this function is called so many times, I didn't know where to solve the problem… in core or in the module.

**Comment by Graham-72** [comment]
Now that we have Backdrop 1.1, I have resumed work on porting Date and tackling the various problems we have found. I am now attempting to avoid and remove some of the additional format options added in Date for Drupal because Backdrop already has good facilities in core. Hopefully I will have some revisions to publish soon.

**Comment by ghost** [comment]
I just posted this, due to an issue with `date_popup_formats()`: https://github.com/backdrop/backdrop-issues/issues/974
